src/openrct2-ui/windows/RideList.cpp

```diff
@@ -172,872 +143,795 @@ static constexpr const rct_string_id page_names[] = {
+        // owing to its use of ride_has_any_track_elements().
+        // This makes sure it's only refreshed every 64 ticks.
+        if (!(gCurrentRealTimeTicks & 0x3f))
+        {
+            RefreshList();
+        }
     }
 
-    // Refreshing the list can be a very intensive operation
-    // owing to its use of ride_has_any_track_elements().
-    // This makes sure it's only refreshed every 64 ticks.
-    if (!(gCurrentRealTimeTicks & 0x3f))
+    /**
+     *
+     *  rct2: 0x006B3511
+     */
+    void OnMouseUp(rct_widgetindex widgetIndex) override
     {
-        window_ride_list_refresh_list(w);
+        switch (widgetIndex)
+        {
+            case WIDX_CLOSE:
+                window_close(this);
+                break;
+            case WIDX_SORT:
+                this->list_information_type = _windowRideListInformationType;
+                this->selected_list_item = -1;
+                RefreshList();
+                break;
+            case WIDX_TAB_1:
+            case WIDX_TAB_2:
+            case WIDX_TAB_3:
+                if (this->page != widgetIndex - WIDX_TAB_1)
+                {
+                    this->page = widgetIndex - WIDX_TAB_1;
+                    this->frame_no = 0;
+                    this->selected_list_item = -1;
+                    if (this->page != PAGE_RIDES && _windowRideListInformationType > INFORMATION_TYPE_RUNNING_COST)
+                    {
+                        _windowRideListInformationType = INFORMATION_TYPE_STATUS;
+                    }
+                    RefreshList();
+                }
+                break;
+            case WIDX_CLOSE_LIGHT:
+                CloseAllRides();
+                break;
+            case WIDX_OPEN_LIGHT:
+                OpenAllRides();
+                break;
+            case WIDX_QUICK_DEMOLISH:
+                if (network_get_mode() != NETWORK_MODE_CLIENT)
+                {
+                    _quickDemolishMode = !_quickDemolishMode;
+                }
+                else
+                {
+                    _quickDemolishMode = false;
+                }
+                this->Invalidate();
+                break;
+        }
     }
-}
 
-/**
- *
- *  rct2: 0x006B3532
- */
-static void window_ride_list_mousedown(rct_window* w, rct_widgetindex widgetIndex, rct_widget* widget)
-{
-    if (widgetIndex == WIDX_OPEN_CLOSE_ALL)
-    {
-        gDropdownItemsFormat[0] = STR_CLOSE_ALL;
-        gDropdownItemsFormat[1] = STR_OPEN_ALL;
-        WindowDropdownShowText(
-            { w->windowPos.x + widget->left, w->windowPos.y + widget->top }, widget->height(), w->colours[1], 0, 2);
-    }
-    else if (widgetIndex == WIDX_INFORMATION_TYPE_DROPDOWN)
+    /**
+     *
+     *  rct2: 0x006B3532
+     */
+    void OnMouseDown(rct_widgetindex widgetIndex) override
     {
-        widget--;
+        if (widgetIndex == WIDX_OPEN_CLOSE_ALL)
+        {
+            auto* widget = &this->widgets[widgetIndex];
```

Comment:
```suggestion
            const auto& widget = this->widgets[widgetIndex];
```
Use a reference over a pointer. Also use const when the data is being read only. This is not the only line, please correct all the cases.